tools/ci-log-clean.py

```
@@ -92,7 +93,93 @@ def clean(text):
+            return [], {"lines": 0, "dropped": 0, "marked": 0, "groups": 0}
+
+    CB = "^[[36;1m"          # what `gh` actually emits — measured: 218 pairs, 0 real ESC
+    ESCB = "\x1b[36;1m"     # a real ESC byte, the form a reader would guess
+
+    print("⛔ the defect this exists for: an echoed script counted as output")
+    log = "\n".join([
+        f"{CB}grep -c FAILED $FAILED_FILES^[[0m",
+        "0",
+    ])
+    out, st = cleaned(log)
+    check("the echoed grep line is dropped", any("FAILED" in l for l in out), False)
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)